code/modules/vehicles/tank.dm

```diff
@@ -0,0 +1,381 @@
+
+/obj/item/tank_weapon/minigun/can_fire(var/turf/T)
+	return TRUE //No loc check here
+
+/obj/vehicle/tank
+	name = "MK-1 'friendly fire' prototype tank"
+	desc = "A gigantic wall of metal designed for maximum Xeno destruction. Click it with an open hand to enter as a pilot or a gunner."
+	icon = 'icons/obj/tank.dmi'
+	icon_state = "tank"
+	layer = OBJ_LAYER
+	bound_width = 128
+	bound_height = 128
+	anchored = FALSE
+	can_buckle = FALSE
+	req_access = list(ACCESS_MARINE_TANK)
+	move_delay = 4
+	obj_integrity = 600
+	max_integrity = 600
+	anchored = TRUE //No bumping / pulling the tank
+	demolish_on_ram = TRUE
+	//Who's driving the tank
+	var/mob/living/carbon/human/pilot
+	var/mob/living/carbon/human/gunner
+	var/list/operators = list() //Who's in this tank? Prevents you from entering the tank again
+	//Health and combat shit
+	var/obj/turret_overlay/turret_overlay //Allows for independantly swivelling guns, wow!
+	var/obj/turret_overlay/minigun/minigun_overlay
+	var/obj/item/tank_weapon/main_cannon //What we use to shoot big shells
+	var/obj/item/tank_weapon/minigun/minigun //What we use to shoot mini shells ((rapidfire xenocrusher 6000))
+	var/main_cannon_dir = null //So that the guns swivel independantly
+	var/minigun_dir = null
+	var/atom/firing_target = null //Shooting code, at whom are we firing?
+	var/firing_main_cannon = FALSE
+	var/firing_minigun = FALSE
+	var/last_drive_sound = 0 //Engine noises.
+
+/obj/vehicle/tank/examine(mob/user)
+	. = ..()
+	to_chat(user, "<b>To fire its main cannon, <i>ctrl</i> click a tile</b>")
+	to_chat(user, "<b>To fire its minigun, click a tile</b>")
+
+/obj/turret_overlay
+	name = "Tank gun turret"
+	desc = "The shooty bit on a tank."
+	icon = 'icons/obj/tank_gun.dmi'
+	icon_state = "turret"
+	layer = ABOVE_OBJ_LAYER
+	animate_movement = TRUE //So it doesnt just ping back and forth and look all stupid
+	mouse_opacity = FALSE //It's an overlay
+
+/obj/vehicle/tank/Initialize()
+	. = ..()
+	turret_overlay = new()
+	update_icon()
+	main_cannon = new(src) //Make our guns
+	minigun = new(src)
+	main_cannon.owner = src
+	minigun.owner = src
+	GLOB.tank_list += src
+
+/obj/vehicle/tank/Destroy()
+	qdel(turret_overlay)
+	qdel(main_cannon)
+	qdel(minigun)
+	. = ..()
+
+/obj/vehicle/tank/Move()
+	. = ..()
+	update_icon()
+	if(world.time > last_drive_sound + 2 SECONDS)
+		playsound(src, 'sound/ambience/tank_driving.ogg', vol = 20, sound_range = 30)
+		last_drive_sound = world.time
+
+/obj/vehicle/tank/update_icon() //To show damage, gun firing, whatever. We need to re apply the gun turret overlay.
+	var/icon/I = icon(icon,icon_state,dir)
+	bound_width = I.Width() //Adjust the hitbox in case admins want to make an OMEGAtank
+	bound_height = I.Height()
+	vis_contents = null
+	if(!turret_overlay) //We have to do this because of BYOND's innate dir setting habits, in byond, if you have an overlay'd object its direction can ONLY be which way the source is facing. So, we improvise.. ((yes this is kinda shit))
```

Comment:
By ninjanomnom: "Comment is wrong, you can make an overlay from an image to get around this bug"